tests/suites/test_suite_pk.function

```diff
@@ -90,13 +90,128 @@ size_t mbedtls_rsa_key_len_func( void *ctx )
     return( ((const mbedtls_rsa_context *) ctx)->len );
 }
 #endif /* MBEDTLS_RSA_C */
+
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#define PK_PSA_INVALID_SLOT 0 /* guaranteed invalid */
+
+/*
+ * Generate a key in a free key slot and return this key slot,
+ * or PK_PSA_INVALID_SLOT if no slot was available.
+ * The key uses NIST P-256 and is usable for signing with SHA-256.
+ */
+psa_key_slot_t pk_psa_genkey( void )
+{
+    psa_key_slot_t key;
+
+    const int curve = PSA_ECC_CURVE_SECP256R1;
+    const psa_key_type_t type = PSA_KEY_TYPE_ECC_KEYPAIR(curve);
+    const size_t bits = 256;
+
+    psa_key_policy_t policy;
+
+    /* find a free key slot
+     * this is temporary, a new PSA API for this is coming */
+    for( key = 1; key <= 32; key++ )
+        if( psa_get_key_information( key, NULL, NULL ) == PSA_ERROR_EMPTY_SLOT )
+            goto found;
+
+    /* not found */
+    return( PK_PSA_INVALID_SLOT );
+
+found:
+    /* set up policy on key slot */
+    psa_key_policy_init( &policy );
+    psa_key_policy_set_usage( &policy, PSA_KEY_USAGE_SIGN,
+                                      PSA_ALG_ECDSA(PSA_ALG_SHA_256) );
+    if( PSA_SUCCESS != psa_set_key_policy( key, &policy ) )
+        return( PK_PSA_INVALID_SLOT );
+
+    /* generate key */
+    if( PSA_SUCCESS != psa_generate_key( key, type, bits, NULL, 0 ) )
+        return( PK_PSA_INVALID_SLOT );
+
+    return( key );
+}
+#endif /* MBEDTLS_USE_PSA_CRYPTO */
 /* END_HEADER */
 
 /* BEGIN_DEPENDENCIES
  * depends_on:MBEDTLS_PK_C
  * END_DEPENDENCIES
  */
 
+/* BEGIN_CASE depends_on:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_ECDSA_C:MBEDTLS_ECP_DP_SECP256R1_ENABLED */
+void pk_psa_utils(  )
+{
+    mbedtls_pk_context pk, pk2;
```

Comment:
We should add a call to `psa_init()` somewhere. Probably in the [main() in main.function](https://github.com/ARMmbed/mbedtls/blob/development/tests/suites/main_test.function#L216) next to the platform initialization?

Reply:
Ok, so my memory wasn't that bad, I was just looking for the wrong thing: it's called `psa_crypto_init()` and I added it in the body of both test functions. Of course I agree it should go in the main test function, so this is a reminder for me to remove the existing occurrences when doing that.